### PR TITLE
BB-102: fix lifecycle rules check issue from BB-83

### PR DESCRIPTION
**Pull request by alexanderchan-scality**

### CODE REVIEW

File: extensions/lifecycle/bucketProcessor/LifecycleBucketProcessor.js [hunk 1]
```diff
@@ -76,6 +77,17 @@ class LifecycleBucketProcessor {
         this._producer = null;
         this._kafkaBacklogMetrics = null;
 
+        this._supportedRulesObject = {};
+        supportedLifecycleRules.forEach(rule => {
+            this._supportedRulesObject[rule] = { enabled: true };
+        });
+        this._hasSupportedRules = Array.isArray(supportedLifecycleRules) &&
```
Comment (jonathan-gramain) [inline extensions/lifecycle/bucketProcessor/LifecycleBucketProcessor.js:81]: Wouldn't we need to also check that at least one of the supported rules has the `enabled: true` flag? Just to maintain the same behavior than before.

File: extensions/lifecycle/bucketProcessor/LifecycleBucketProcessor.js [hunk 2]
```diff
@@ -76,6 +77,17 @@ class LifecycleBucketProcessor {
...
+            supportedLifecycleRules.length > 0;
```
Comment (tmacro) [inline extensions/lifecycle/bucketProcessor/LifecycleBucketProcessor.js]: If `supportedLifecycleRules` can be something other than an array I think we should guard the `forEach` as well
```suggestion
        this._hasSupportedRules = Array.isArray(supportedLifecycleRules) &&
            supportedLifecycleRules.length > 0;
        this._supportedRulesObject = {};
        if (this._hasSupportedRules) {
            supportedLifecycleRules.forEach(rule => {
                this._supportedRulesObject[rule] = { enabled: true };
            });
        }
```

Comment (alexanderchan-scality) [reply extensions/lifecycle/bucketProcessor/LifecycleBucketProcessor.js]: `supportedLifecycleRules` is a constant from [arsenal](https://github.com/scality/Arsenal/blob/development/8.1/lib/constants.js#L146); so it safe to say that it is always an array, but I see you point

File: extensions/lifecycle/bucketProcessor/LifecycleBucketProcessor.js (hunk 1 above)
Comment (alexanderchan-scality) [reply extensions/lifecycle/bucketProcessor/LifecycleBucketProcessor.js:81]: I don't believe so. the `supportedLifecycleRules` will be a constant from arsenal, and the generated object will always have `enabled: true` for each key.

**Review by jonathan-gramain — APPROVED** [review]

**Review by tmacro — APPROVED** [review]

**Pull request merged**